sn_client/src/uploader/upload.rs

```diff
@@ -824,22 +824,42 @@ impl InnerUploader {
             let mut cost_map = BTreeMap::new();
             let mut current_batch = vec![];
 
+            let mut got_a_previous_force_payment = false;
             while let Some(payment) = make_payment_receiver.recv().await {
                 let make_payments = if let Some((item, quote)) = payment {
                     let xorname = item.xorname();
                     trace!("Inserted {xorname:?} into cost_map");
 
                     current_batch.push((xorname, quote.clone()));
                     let _ = cost_map.insert(xorname, (quote.1, quote.2, quote.0.to_bytes()));
-                    cost_map.len() >= batch_size
+                    cost_map.len() >= batch_size || got_a_previous_force_payment
                 } else {
                     // using None to indicate as all paid.
```

Comment:
better update the comment to be : 
```
When the received `payment` is `None`, it notifies no more payment to be made.
There shall be a round of `forced payment` to pay the remaining payments in the cost_map (less than batch_size)
With normal order, there shall be no more payment come in afterwards,
however it could be break due to async channel send.
Hence have to catering such `out-of-order` situation.
```